src/components/views/elements/Validation.tsx

```diff
@@ -111,13 +108,13 @@ export default function withValidation<T = void, D = void>({
                     continue;
                 }
 
-                if (rule.skip?.call(this, data, derivedData)) {
+                if (rule.skip?.call(this, data, derivedData!)) {
```

Comment:
Why is `derivedData` set here?
Maybe in non-obvious cases we should add comments?

Reply:
The issue is actually higher up, `derivedData` will always be of the right type, its just that the type is sometimes `undefined` - I couldn't find an ideal way to express it in the generic. If `deriveData === undefined then D is undefined, else D is something else`

Reply:
Can do it with a cast on the set instead, much better